### ISSUE TITLE
Script batch compilation/evaluation

### ISSUE THREAD

**Issue by AndriySvyryd** [dotnet/roslyn#21525]
In my app there are hundreds of scripts that need to be evaluated at startup. This allocates tens of gigabytes that need to be GC'ed, resulting in a very noticeable delay.
Is there an efficient way of compiling and/or evaluating all the scripts at once?

**Comment by tmat** [comment]
Do you mean compile them ahead of running the app into a binary on disk?

**Comment by AndriySvyryd** [comment]
No, they need to be compiled at runtime. All of them use the same references and imports and could share the same compilation unit.

**Comment by AndreasHeisel** [comment]
Hello Rosly-Team,

we have a similar scenario: Many scripts that could be compiled together in one assembly to speed things up. 
There is no need for a binary on disc. We would compile the scripts during app startup and sometimes as batches on runtime.

This would speed up not only app startup but also runtime. See dotnet/runtime#57652

I could image something like this:
```cs
public class Script
{
  ...
  // diagnostics and script are matched by index: first script => first diagnostics etc.
  public static IEnumerable<ImmutableArray<Diagnostic>> BatchCompile(IEnumerable<Script> scripts, CancellationToken cancellationToken = default);

  // or as alternative
  // the dictionary values will be filled during compilation
  public static void BatchCompile(IDictionary<Script, ImmutableArray<Diagnostic>> scripts, CancellationToken cancellationToken = default);
  ...
}
```
